### PR TITLE
[MBL-15428][Teacher] Teacher edit dashboard

### PR DESCRIPTION
**Pull request by kristofnemere**
refs: MBL-15428
affects: Teacher
release note: Implemented edit dashboard functionality.

test plan: in the ticket, smoke test Student dashboard.

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [x] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/EditDashboardPageTest.kt [hunk 1]
```diff
@@ -26,48 +26,45 @@ import dagger.hilt.android.testing.HiltAndroidTest
...
-        editCoursesListPage.assertHasCourses(courses)
+        val course = data.courses.values.toList()[0]
+        coursesListPage.openEditDashboard()
+        editDashboardPage.assertHasCourse(course.name)
     }
 
     @Test
     @TestRail(ID = "C3109574")
-    fun favoriteCourse() {
+    fun favourCourse() {
```
Comment (tamaskozmer) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/EditDashboardPageTest.kt:50]: I think this is a little bit strange wording, I never heard this in a context like this.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/DashboardFragment.kt [hunk 2]
```diff
@@ -15,53 +15,70 @@
...
 import org.greenrobot.eventbus.EventBus
 import org.greenrobot.eventbus.Subscribe
 import org.greenrobot.eventbus.ThreadMode
+import javax.inject.Inject
 
 private const val LIST_SPAN_COUNT = 1
 
 @ScreenView(SCREEN_VIEW_DASHBOARD)
-class CoursesFragment : BaseSyncFragment<Course, CoursesPresenter, CoursesView, CoursesViewHolder, CoursesAdapter>(), CoursesView {
+@AndroidEntryPoint
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/fragments/DashboardFragment.kt:58]: I don't think this needs to be an `@AndroidEntryPoint`.

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/DashboardFragment.kt [hunk 3]
```diff
@@ -15,53 +15,70 @@
...
+class DashboardFragment : BaseSyncFragment<Course, CoursesPresenter, CoursesView, CoursesViewHolder, CoursesAdapter>(), CoursesView {
+
+    @Inject
+    lateinit var editDashboardRouter: EditDashboardRouter
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/fragments/DashboardFragment.kt:62]: This is not used here.

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/DashboardFragment.kt [hunk 4]
```diff
@@ -118,28 +137,20 @@ class CoursesFragment : BaseSyncFragment<Course, CoursesPresenter, CoursesView,
...
-            NoInternetConnectionDialog.show(requireFragmentManager())
-        }
+        mNeedToForceNetwork = false
     }
 
     private fun setupHeader() {
-        seeAllTextView.setTextColor(ThemePrefs.buttonColor)
-        seeAllTextView.setOnClickListener { mCourseListCallback?.onShowAllCoursesList() }
+        editDashboardTextView.setTextColor(ThemePrefs.buttonColor)
+        editDashboardTextView.setOnClickListener { mCourseListCallback?.onEditDashboard() }
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/fragments/DashboardFragment.kt:153]: I don't think this callback is necessary anymore. There is no point propagating this call to the Activty, we can just call `RouteMatcher.route(requireActivity, EditDashboardFragment.makeRoute())` instead and get rid of this callback completely.

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/dashboard/edit/EditDashboardViewModel.kt [hunk 5]
```diff
@@ -15,28 +15,28 @@
  */
 
 
-package com.instructure.student.features.dashboard.edit
+package com.instructure.pandautils.features.dashboard.edit
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/dashboard/edit/EditDashboardViewModel.kt:18]: I think there we should add some different behaviors for this class whether we are in the student app or in the teacher. Let's discuss this on Slack.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/AllCoursesFragment.kt [hunk 6]
```diff
@@ -1,192 +0,0 @@
...
-        val padding = resources.getDimensionPixelSize(R.dimen.course_list_padding)
-        mRecyclerView.setPaddingRelative(padding, padding, padding, padding)
-        mRecyclerView.clipToPadding = false
-    }
-
-    override fun onReadySetGo(presenter: AllCoursesPresenter) {
-        if(mRecyclerView.adapter == null) {
-            mRecyclerView.adapter = adapter
-        }
-        presenter.loadData(false)
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/fragments/AllCoursesFragment.kt:97]: Please also remove the Presenter and the PresenterFactory.

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/EditFavoritesFragment.kt [hunk 7]
```diff
@@ -1,146 +0,0 @@
...
-            } else {
-                presenter.refresh(true)
-            }
-        }
-
-        mRecyclerView.layoutManager = mLayoutManager
-        addSwipeToRefresh(swipeRefreshLayout)
-        mRecyclerView.adapter = adapter
-
-        presenter.loadData(true)
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/fragments/EditFavoritesFragment.kt:83]: Please remove the Presenter and the PresenterFactory as well.

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/EditDashboardPageTest.kt (hunk 1 above)
Comment (kdeakinstructure) [reply apps/teacher/src/androidTest/java/com/instructure/teacher/ui/EditDashboardPageTest.kt:50]: We thought that as well but the Android Studio suggested this change and it does not underlines it this way...

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by stephaniegilbert944 — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/main/java/com/instructure/teacher/presenters/CoursesPresenter.kt [hunk 8]
```diff
@@ -61,12 +62,12 @@ class CoursesPresenter : SyncPresenter<Course, CoursesView>(Course::class.java)
         apiAsync<List<DashboardCard>> { CourseManager.getDashboardCourses(forceNetwork, it) }.await()
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/presenters/CoursesPresenter.kt:62]: You might as well rename this class if the Fragment was also renamed to Dashboard.

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/dashboard/edit/EditDashboardViewModel.kt [hunk 9]
```diff
@@ -282,7 +282,7 @@ class EditDashboardViewModel @Inject constructor(
     }
 
     private fun selectAllCourses() {
-        val coursesToFavorite = (currentCoursesViewData + futureCoursesViewData).filter { !it.isFavorite && it.favoriteable }
+        val coursesToFavorite = (currentCoursesViewData + pastCoursesViewData + futureCoursesViewData).filter { !it.isFavorite && it.favoriteable }
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/dashboard/edit/EditDashboardViewModel.kt:285]: What is the reason for this change?

File: libs/pandautils/src/test/java/com/instructure/pandautils/features/dashboard/edit/EditDashboardViewModelTest.kt [hunk 10]
```diff
@@ -844,23 +768,21 @@ class EditDashboardViewModelTest {
     fun `Deleted courses cannot be opened`() {
```
Comment (tamaskozmer) [inline libs/pandautils/src/test/java/com/instructure/pandautils/features/dashboard/edit/EditDashboardViewModelTest.kt:768]: I think this test and the test above is the same, one of them should be removed and the other maybe renamed.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Review by tamaskozmer — APPROVED** [review]

**Pull request merged**